respy/simulate.py

```diff
@@ -218,10 +219,13 @@ def simulate(
             optim_paras=optim_paras,
         )
 
-        data.append(current_df_extended)
+        data.append(current_df_extended.copy(deep=True))
 
         if is_n_step_ahead and period != n_simulation_periods - 1:
-            df = _apply_law_of_motion(current_df_extended, optim_paras)
```

Comment:
Would it be an option to not apply the law of motion here but take choice and dense key to get the next indices directly from the state space? 
Then we would also save the indexer step and save the calculation.

Reply:
I have created the issue #376 to discuss your point. Even if my point on cpu bound and io bound is not true, it makes sense to separate the changes. Is it ok for you?